sympy/polys/rings.py

```diff
@@ -2193,7 +2195,10 @@ def _gcd(f, g):
             return ring.dmp_inner_gcd(f, g)
 
     def _gcd_ZZ(f, g):
-        return heugcd(f, g)
+        h = _gcd_prs(f, g)
+        cff = f.div(h)[0]
+        cfg = g.div(h)[0]
+        return h, cff, cfg
```

Comment:
I am not sure yet that we would want to keep this change at the point when this PR is merged but it is good to keep it there for now so that the prs method can be tested while the PR is in development.